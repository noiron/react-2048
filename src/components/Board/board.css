.boardTable {
    margin: 10px 0;
}

.boardRow {
    height: 100px;
    display: flex;
    margin-bottom: 6px;
}
