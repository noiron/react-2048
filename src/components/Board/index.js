import React from 'react';
import PropTypes from 'prop-types';
import Row from '../Row';
import './board.css';

export default class Board extends React.Component {
    static propTypes = {
        matrix: PropTypes.arrayOf(PropTypes.array).isRequired
    };

    render() {
        const { matrix } = this.props;

        return (
            <table>
                <tbody>
                    {matrix.map((row, index) => <Row row={row} key={index} />)}
                </tbody>
            </table>
        );
    }
}
