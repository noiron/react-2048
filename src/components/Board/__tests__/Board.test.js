import React from 'react'
import renderer from 'react-test-renderer';
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import Board from '..';

Enzyme.configure({ adapter: new Adapter() });

describe('<Board />', () => {
    it('component render', () => {
        const matrix = [[0, 0, 4, 4], [0, 2, 0, 0], [4, 0, 0, 0], [0, 2, 0, 0]];
        const board = renderer.create(<Board matrix={matrix} />).toJSON();
        expect(board).toMatchSnapshot();
    })
})
