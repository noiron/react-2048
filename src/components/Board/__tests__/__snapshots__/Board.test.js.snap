// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Board /> component render 1`] = `
<div
  className={undefined}
>
  <div
    className={undefined}
  >
    <div
      className=" "
    >
      <div
        className={undefined}
      />
    </div>
    <div
      className=" "
    >
      <div
        className={undefined}
      />
    </div>
    <div
      className=" "
    >
      <div
        className={undefined}
      >
        4
      </div>
    </div>
    <div
      className=" "
    >
      <div
        className={undefined}
      >
        4
      </div>
    </div>
  </div>
  <div
    className={undefined}
  >
    <div
      className=" "
    >
      <div
        className={undefined}
      />
    </div>
    <div
      className=" "
    >
      <div
        className={undefined}
      >
        2
      </div>
    </div>
    <div
      className=" "
    >
      <div
        className={undefined}
      />
    </div>
    <div
      className=" "
    >
      <div
        className={undefined}
      />
    </div>
  </div>
  <div
    className={undefined}
  >
    <div
      className=" "
    >
      <div
        className={undefined}
      >
        4
      </div>
    </div>
    <div
      className=" "
    >
      <div
        className={undefined}
      />
    </div>
    <div
      className=" "
    >
      <div
        className={undefined}
      />
    </div>
    <div
      className=" "
    >
      <div
        className={undefined}
      />
    </div>
  </div>
  <div
    className={undefined}
  >
    <div
      className=" "
    >
      <div
        className={undefined}
      />
    </div>
    <div
      className=" "
    >
      <div
        className={undefined}
      >
        2
      </div>
    </div>
    <div
      className=" "
    >
      <div
        className={undefined}
      />
    </div>
    <div
      className=" "
    >
      <div
        className={undefined}
      />
    </div>
  </div>
</div>
`;
